src/BeamAdapter/component/BeamInterpolation.inl

```diff
@@ -137,36 +203,13 @@ void BeamInterpolation<DataTypes>::bwdInit()
         return;
 
     Size nbEdges = this->m_topology->getNbEdges();
-    auto youngModulus = sofa::helper::getWriteOnlyAccessor(d_defaultYoungModulus);
-    if (youngModulus.size() != nbEdges)
-    {
-        Real value = m_defaultYoungModulus;
-        if (youngModulus.size() == 0){
-            msg_warning() << "Empty data field for " << d_defaultYoungModulus.getName() <<". Set default " << value;
-        } else {
-            value = youngModulus[0];
-        }
-        youngModulus.resize(nbEdges);
-        for (auto& beamYoungModulus: youngModulus)
-            beamYoungModulus = value;
-    }
-    m_defaultYoungModulus = youngModulus[0]; // if the sizes mismatch again at runtime, will use this default value
-
-    auto poissonRatio = sofa::helper::getWriteOnlyAccessor(d_poissonRatio);
-    if (poissonRatio.size() != nbEdges)
-    {
-        Real value = m_defaultPoissonRatio;
-        if (poissonRatio.size() == 0){
-            msg_warning() << "Empty data field for " << d_poissonRatio.getName() << ". Set default " << value;
-        } else {
-            value = poissonRatio[0];
-        }
-        poissonRatio.resize(nbEdges);
-        for (auto& beamPoissonRatio: poissonRatio)
-            beamPoissonRatio = value;
-    }
-    m_defaultPoissonRatio = poissonRatio[0]; // if the sizes mismatch again at runtime, will use this default value
-
+    checkDataSize(m_defaultRadius, d_radius, nbEdges);
+    checkDataSize(m_defaultInnerRadius, d_innerRadius, nbEdges);
+    checkDataSize(m_defaultLengthY, d_lengthY, nbEdges);
+    checkDataSize(m_defaultLengthZ, d_lengthZ, nbEdges);
+    checkDataSize(m_defaultYoungModulus, d_defaultYoungModulus, nbEdges);
+    checkDataSize(m_defaultPoissonRatio, d_poissonRatio, nbEdges);

```

Comment:
These calls change the `m_defaultvalues` based on the `d_values` set by the user.

See line 193 :   
```cpp
defaultValue = values[0]; // if the sizes mismatch again at runtime, will use this default value
```